File: com.systemincloud.examples.tasks.javatask/src/test/java/tasks/data/DataDecimalDims.java
package tasks.data;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

import com.systemincloud.modeler.tasks.javatask.api.InputPort;
import com.systemincloud.modeler.tasks.javatask.api.JavaTask;
import com.systemincloud.modeler.tasks.javatask.api.OutputPort;
import com.systemincloud.modeler.tasks.javatask.api.annotations.InputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.JavaTaskInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.OutputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.data.Decimal;

@JavaTaskInfo
public class DataDecimalDims extends JavaTask {

	@InputPortInfo(name = "In", dataType = Decimal.class)
	public InputPort in;
	@OutputPortInfo(name = "Out", dataType = Decimal.class)
	public OutputPort out;

	@Override
	public void execute(int grp) {
		Decimal inDecimal = in.getData(Decimal.class);
		List<BigDecimal> outdecimal = new ArrayList<>(inDecimal.getNumberOfElements());
		int i = 0;
		for(BigDecimal v : inDecimal.getValues()) {
			outdecimal.add(v.movePointRight(i));
			i++;
		}
		out.putData(new Decimal(inDecimal.getDimensions(), outdecimal));
	}

}

File: com.systemincloud.examples.tasks.pythontask/src/test/py/tasks/data/DataDecimalDims.py
from sicpythontask.PythonTaskInfo import PythonTaskInfo
from sicpythontask.PythonTask import PythonTask
from sicpythontask.InputPort import InputPort
from sicpythontask.OutputPort import OutputPort
from sicpythontask.data.Decimal import Decimal
from decimal import Decimal as D

@PythonTaskInfo
class DataDecimalDims(PythonTask):

    def __init_ports__(self):
        self.in_ = InputPort(name="in", data_type=Decimal)
        self.out = OutputPort(name="out", data_type=Decimal)

    def execute(self, grp):
        in1Data = self.in_.get_data(Decimal)
        dec = list(map(lambda x:x* D(10), in1Data.values))
        self.out.put_data(Decimal(dec, in1Data.dims))
